pkgs/applications/networking/cluster/terraform-providers/update-provider

```diff
@@ -1,30 +1,79 @@
 #!/usr/bin/env nix-shell
-#! nix-shell -i bash -p coreutils curl jq
+#! nix-shell -i bash -p coreutils curl jq moreutils
 # shellcheck shell=bash
+# vim: ft=sh
+#
+# Update a terraform provider to the latest version advertised at the
+# provider source address.
 #
-# Update a terraform provider to the latest version advertised at
-# the provider source address.
 set -euo pipefail
 
-USAGE=$(cat<<DOC
-Specify the terraform provider name to update.
+show_usage() {
+  cat <<DOC
+Usage: ./update-provider [--force] [--vendor] <provider_name>
+
+Update a single provider with name 'provider_name' in the providers.json
+inventory file.
 
 Example:
 To update nixpkgs.terraform-providers.aws run:
 ./update-provider aws
+
+Options:
+* --force: Force the update even if the version matches.
+* --vendor: Switch from go package to go modules with vendor.
 DOC
-)
+}
+
+force=
+provider_name=
+vendor=
+
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+  -h | --help)
+    show_usage
+    exit
+    ;;
+  --force)
+    force=1
+    shift
+    ;;
+  --vendor)
+    vendor=1
+    shift
+    ;;
+  *)
+    if [[ -n "$provider_name" ]]; then
+      echo "ERROR: provider name was passed two times: '$provider_name' and '$1'"
+      echo "Use --help for more info"
+      exit 1
+    fi
+    provider_name=$1
+    shift
+  esac
```

Comment:
You can do one shift at the end.

Reply:
I find it generally clearer to deal with shifting per argument as the number of shifts might vary per flag (not here though). YMMV.